core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -9004,6 +9005,38 @@ LogicalProject(EXPR$0=[$1], EXPR$1=[$0])
           LogicalTableScan(table=[[CATALOG, SALES, EMP]])
         LogicalProject(ENAME=[$0])
           LogicalTableScan(table=[[CATALOG, SALES, BONUS]])
+]]>
+    </Resource>
+  </TestCase>
+  <TestCase name="testPushProjectPastLeftJoin2">
+    <Resource name="sql">
+      <![CDATA[select coalesce(d.name, b.job, '')
+from emp e
+left join bonus b on e.ename = b.ename
+left join dept d on e.deptno = d.deptno]]>
+    </Resource>
+    <Resource name="planBefore">
+      <![CDATA[
+LogicalProject(EXPR$0=[CASE(IS NOT NULL($14), CAST($14):VARCHAR(10) NOT NULL, IS NOT NULL($10), CAST($10):VARCHAR(10) NOT NULL, '':VARCHAR(10))])
+  LogicalJoin(condition=[=($7, $13)], joinType=[left])
+    LogicalJoin(condition=[=($1, $9)], joinType=[left])
+      LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+      LogicalTableScan(table=[[CATALOG, SALES, BONUS]])
+    LogicalTableScan(table=[[CATALOG, SALES, DEPT]])
+]]>
+    </Resource>
+    <Resource name="planAfter">
+      <![CDATA[
+LogicalProject(EXPR$0=[CASE(IS NOT NULL($4), CAST($4):VARCHAR(10) NOT NULL, $1, $2, '':VARCHAR(10))])
+  LogicalJoin(condition=[=($0, $3)], joinType=[left])
+    LogicalProject(DEPTNO=[$1], EXPR$0=[IS NOT NULL($3)], EXPR$1=[CAST($3):VARCHAR(10) NOT NULL])
```

Comment:
@wojustme If the $3 is NULL, then what will happen in `EXPR$1=[CAST($3):VARCHAR(10) NOT NULL])`? Because the EXPR1 will execute any time. right? 
But  the plan bafore, when we confirm $3 is not NULL, we will to calcute `[CAST($3):VARCHAR(10) NOT NULL])`,
So please try when job include NULL value not just claim this filed can NULL, Plase check the data, I don't know, Just want to confirm.

Reply:
@NobiGo 
It's good question for me. 
In my project, we use calcite as an sql-proxy, and send dialectal sql to various query engine.
`EXPR$1=[CAST($3):VARCHAR(10) NOT NULL])` will be translated to sql fragment of `CAST(job AS VARCHAR)`
So, I didn't pay attention to this problem. I try to find some codes or docs to explain this question.

In calcite's linq4j, `CAST($3):VARCHAR(10) NOT NULL` will be translated to `input_isNull0 ? (String) null : input_value0`

Reply:
Hi, Please try validate this SQL result in out.iq and submit the test in this PR? I don't know.